### ISSUE TITLE
trailing comma required in call to single-param method

### ISSUE THREAD

**Issue by alexeagle** [palantir/tslint#1551]
In this example, we have a multi-line string literal passed as the argument to a single-param function:

```
function stripTemplate(i:string) {
  return i.replace(/_/g, "").replace(/\n/g, "").replace(/<br>/g, "\n");
}
stripTemplate(
`{{ initialOffset ? offset()+'{' : '{' }}<br>
_<template ngFor let-f [ngForOf]="fields">
__</template>
{{ offset()+'}' }}`  /* trailing comma is required here?? */
  );
```

TSLint with `{trailing-comma: [true, {"multiline": "always"}]}` says that a trailing comma is required on the line before the closing paren, which looks rather silly.

Now that we have the type-checker we could do better, perhaps.

**Comment by JKillian** [comment]
cc @markwongsk 

@alexeagle, I'm hesitant to make non-type checker rules now require the type-checker. In this case, it feels like overkill.

I think it actually makes sense to require a comma here, it's parallel to other constructs:

```
doSomething(
    someLongArgument * expression - whoKnows,
);
```

**Comment by markwongsk** [comment]
I agree with @JKillian. To elaborate, consider a modification of your example:
```
stripTemplate(
    s /* should flag trailing comma required */
);

stripTemplate(
    "hello" + "world" /* also here */
);

stripTemplate(
    "hello" +
    "world" /* also here */
);

stripTemplate(
    myFunctionCall() /* also here */
);
```
There's nothing the typechecker is going to give than the syntax tree doesn't already (in this case LEFT_PAREN, SYNTAX_LIST, RIGHT_PAREN). The linter lints the syntax list and flags if a comma isn't present.

**Comment by alexeagle** [comment]
I'm sold on not requiring the type-checker for existing rules. And we didn't make it possible for a rule to use the type-checker if present, and carry on with syntax only if the type checker is not present, so we couldn't hint at it.

Also, if I clang-format this example, it moves the closing paren to the previous line. So I think my root problem here was that this code example wasn't well formatted.
